tests/Hamcrest/File/IsWritableFileTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * @author Vasek Brychta <vaclav@brychtovi.cz>
+ */
+
+namespace Hamcrest\File;
+
+use Hamcrest\AbstractMatcherTest;
+
+class IsWritableFileTest extends AbstractMatcherTest
+{
+	private static $WRITABLE_FILE_PATH;
+
+	private static $WRITABLE_FILE;
+	private static $NON_WRITABLE_FILE;
+
+	/**
+	 * @beforeClass
+	 */
+	public static function initializeFiles()
+	{
+		self::$WRITABLE_FILE_PATH = __FILE__;
+
+		self::$WRITABLE_FILE = new \SplFileInfo(self::$WRITABLE_FILE_PATH);
+
+		self::$NON_WRITABLE_FILE = new class('/tmp/non-writable') extends \SplFileInfo {
```

Comment:
1. What kind of class instantiation syntax is this?
2. Does it create an anonymous class instance?
3. Will this code work in all supported by this library PHP versions?